IThenticate.inc.php

```diff
@@ -0,0 +1,455 @@
+                [
+                    'headers' => $this->getRequiredHeaders(),
+                    'verify' => false,
+                    'exceptions' => false,
+                    'http_errors' => false,
+                ]
+            );
+        } catch (\GuzzleHttp\Exception\ConnectException $exception) {
+            return false;
+        } catch (\Throwable $exception) {
+            throw $exception;
+        }
+
+        return $response->getStatusCode() === 200;
+    }
+
+    /**
+     * Confirm the EULA on the user's behalf for given version
+     * @see https://developers.turnitin.com/docs/tca#accept-eula-version
+     * 
+     * @param User      $user
+     * @param Context   $context
+     *
+     * @return bool
+     */
+    public function confirmEula($user, $context) {
+        
+        $response = Application::get()->getHttpClient()->request(
+            'POST',
+            $this->getApiPath("eula/{$this->getApplicableEulaVersion()}/accept"),
+            [
+                'headers' => array_merge($this->getRequiredHeaders(), [
+                    'Content-Type' => 'application/json'
+                ]),
+                'json' => [
+                    'user_id' => $user->getId(),
+                    'accepted_timestamp' => \Carbon\Carbon::now()->toIso8601String(),
+                    'language' => $this->getEulaConfirmationLocale($context->getPrimaryLocale()),
+                ],
+                'verify' => false,
+                'exceptions' => false,
+            ]
+        );
+
+        return $response->getStatusCode() === 200;
+    }
+    
+    /**
+     * Create a new submission at service's end
+     * @see https://developers.turnitin.com/docs/tca#create-a-submission
+     * 
+     * @param Submission    $submission
+     * @param User          $user
+     * @param Author|null   $author
+     * @param Site|null     $site
+     *
+     * @return string|null  if succeed, it will return the created submission UUID at service's end and 
+     *                      at failure, will return null
+     */
+    public function submitSubmission($submission, $user, $author = null, $site = null) {
+
+        $publication = $submission->getCurrentPublication(); /** @var Publication $publication */
+        $author ??= $publication->getPrimaryAuthor();
+
+        if (!$site) {
+            import('lib.pkp.classes.db.DAORegistry');
+            $siteDao = DAORegistry::getDAO("SiteDAO"); /** @var SiteDAO $siteDao */
+            $site = $siteDao->getSite(); /** @var Site $site */
+        }
+
+        $response = Application::get()->getHttpClient()->request(
+            'POST',
+            $this->getApiPath("submissions"),
+            [
+                'headers' => array_merge($this->getRequiredHeaders(), [
+                    'Content-Type' => 'application/json'
+                ]),
+                'json' => [
+                    'owner' => $author->getId(),
```

Comment:
> email addresses as unique account identifiers in OJS; maybe those would be more appropriate for users; we can't always count on having those for authors, though

That was my initial approach also but because an author may not have an email address and author/owner details is required, we could not use email 

> If this is something we can use for our own arbitrary purposes

Not actually . According to TAC developer doc, submission author/owner unique system identifier is required . The submitter information is optional but in that case owner/author must accept/confirm EULA . As in our case, the author/owner and submitter can be different and it's the submitter who confirms the EULA, we also need to pass it . 

> it might be worth adding some kind of namespacing to avoid confusion -- e.g. `author/54` and `user/54`

I think that a good idea and will do it .